RULES/cc/cc.go

```diff
@@ -92,33 +106,75 @@ type Dep interface {
 type Library struct {
 	Out           core.OutPath
 	Srcs          []core.Path
+	Blobs         []core.Path
```

Comment:
I don't really like this, but I guess it's necessary?

Reply:
Pretty much. The thing is that ObjectFile isn't flexible enough to support blobs, and that's why cc.Library had to take []core.Path as Objs. But obviously that doesn't work when we have multiple toolchains and multiple outputs. ObjectFiles are a bit useless now, especially because you can't access out(). They are basically only really used by Libraries and Binaries internally.

I do have some ideas on how to make that better (I think ObjectFile should be an interface like cc.Dep. I also think that WithToolchain should act on those interfaces, and not on the concrete structs, so it can support user-defined cc.Deps and ObjectFiles), but this works for now.

Reply:
Why necessary? Could we add this as `Objs` or `Deps`?

Reply:
I put blobs on the same level as Srcs here (even though it's arguably much more niche), i.e. source files that can be compiled into .o files.

Reply:
And since there is no generic interface for that yet, I had to to it this way

Reply:
Sorry Lukasz I didn't see your comment. Objs takes a []OutPath, which is not good for multiple toolchains (also I think it's not really a good pattern, it loses the type information). Deps would work, but a Dep wants a CcLibrary(). The blob is compiled as an object file. I can look into maybe compiling it as a library, but I still think we want a way to arbitrarily add a .o file that is compiled by a custom rule, like cc.Dep for libraries.